internal/cluster/refresh_status_preprocessor.go

```diff
@@ -89,7 +90,23 @@ func (r *refreshPreprocessor) preprocess(ctx context.Context, c *clusterPreproce
 	for _, condition := range r.conditions {
 		stateMachineInput[condition.id.String()] = condition.fn(c)
 	}
-	return stateMachineInput, validationsOutput, nil
+	sortedValidationsOutput := sortValidationsOutput(validationsOutput)
+
+	return stateMachineInput, sortedValidationsOutput, nil
+}
+
+func sortValidationsOutput(validationsOutput map[string][]ValidationResult) map[string][]ValidationResult {
```

Comment:
You are sorting the categories and not the results, the problem is with `ValidationResult` the results.

Reply:
Please add UT as well